src/lib/util/arrays.ts

```diff
@@ -12,3 +15,16 @@ function chunk<T>(array: T[], size: number): T[][] {
 function chunkString(str: string, size: number): string[] {
   return chunk([...str], size).map((c) => c.join(''));
 }
+
+function zip<T, S>(a: T[], b: S[]) {
+  assert(a.length === b.length, 'zip(): arrays of unequal length');
```

Comment:
AFAIK zip is usually implemented to take the minimum length when the lists are unequal. I think we should default to that unless there's a reason that wouldn't make sense here.